### DISCUSSION TITLE
フロントエンドのホスティング先をどうするか？

### DISCUSSION THREAD

**Discussion by DaisukeShinoku** [yuki-snow1823/diary#43] [Ideas]
vercelでチーム開発をする際には、少なくとも１人はProプランに加入する必要性が判明した。
（Hobbyのままだと管理者以外のPRではVercelデプロイが失敗してしまう）

[該当のPR](https://github.com/yuki-snow1823/diary/pull/36)
[参考Zenn](https://zenn.dev/ttskch/articles/691fb62fbb6b1b)

今（2023-06-17 11:30）は、堀越さんが [Netlify](https://www.netlify.com/) にシュッとデプロイを試してくれているが、それがうまくいかなかったり、他に選択肢があればアイディアが欲しいお気持ち。

希望条件としては
・完全無料
・チーム開発できる（管理者以外のPUSH、PR作成でも問題が発生しない）

**Comment by yuki-snow1823** [comment]
ありがとうございます！

https://ralacode.com/blog/post/how-to-deploy-react-app-to-netlify/

今こちらをみながらやってます〜

**Comment by yuki-snow1823** [comment]
@DaisukeShinoku 
https://lovely-buttercream-3a7f81.netlify.app/

**Reply by yuki-snow1823** [reply]
https://diary-bice-omega.vercel.app/sample

**Reply by DaisukeShinoku** [reply]
雑にリプ
https://qiita.com/MyPoZi/items/b6b75c98f117f6c66b89

**Reply by DaisukeShinoku** [reply]
やったか・・・？
https://qiita.com/Tori-chan/items/228e39630d150930f53f

**Comment by yuki-snow1823** [comment]
とりあえずnetlifyにデプロイはできた
```
12:22:07 PM: ​
12:22:07 PM: $ yarn build
12:22:07 PM: yarn run v1.22.19
12:22:07 PM: $ react-scripts build && echo '/* /index.html  200' | cat >build/_redirects 
12:22:08 PM: Creating an optimized production build...
12:22:09 PM: Failed to compile.
12:22:09 PM: 
12:22:09 PM: Module not found: Error: You attempted to import /opt/build/repo/frontend/node_modules/@babel/runtime/helpers/esm/taggedTemplateLiteral.js which falls outside of the project src/ directory. Relative imports outside of src/ are not supported.
12:22:09 PM: You can either move it inside src/, or add a symlink to it from project's node_modules/.
12:22:09 PM: error Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
12:22:09 PM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
12:22:09 PM: ​
12:22:09 PM: "build.command" failed   
```

MUIのリリースでエラー

**Comment by yuki-snow1823** [comment]
https://lovely-buttercream-3a7f81.netlify.app/

にデプロイできました！
毎月リミットはありますが、誰がデプロイしてもOKです〜
